tools/meteor-services/deploy.js

```diff
@@ -321,6 +323,206 @@ function canonicalizeSite(site) {
   return parsed.hostname;
 };
 
+// Executes the poll to check for deployment success and outputs proper messages
+// to user about the status of their app during the polling process
+function executePollForDeploymentSuccess(versionId, deployPollTimeout, result) {
+  // Create a default polling configuration for polling for deploy / build
+  // In the future, we may change this to be user-configurable or smart
+  // The user can only currently configure the polling timeout via a flag
+  let pollingConfiguration = new PollingConfiguration(deployPollTimeout);
+  
+  var deploymentPollResult = buildmessage.enterJob({
+    title: "Waiting for Deploy to succeed..."}, 
+    function() {
+      return pollForDeploy(pollingConfiguration,
+        versionId,
+        true);
+    });
+
+  if(deploymentPollResult && deploymentPollResult.isActive) {
+    // The deploy succeeded and the deployed version is now active
+    const {newVersionNum, hostname, username, region, galaxyUrl} = result.payload;
+    var successfulDeploymentMessage = `
+    ******************************************************************************
+    You have successfully upgraded your app to version ${newVersionNum}.
+    Hostname: ${hostname}, Region: ${region}, Username: ${username}
+    To see the status of your app, visit your app's page on Galaxy
+    \t\t${galaxyUrl}.
+    ******************************************************************************
+    `
+    Console.info(successfulDeploymentMessage);
+  } else {
+    // The deploy did not succeed or the polling timed out
+    if(deploymentPollResult.buildStatus === 'failed-permanently' ||
+      deploymentPollResult.deployStatus === 'failed') {
+        const {newVersionNum, hostname, username, region, galaxyUrl} = result.payload;
+        var failedDeploymentMessage = `
+        ******************************************************************************
+        Failed to deploy your app to version ${newVersionNum}. No changes were made.
+        Hostname: ${hostname}, Region: ${region}, Username: ${username}
+        To see the logs for your app, visit ${galaxyUrl}/logs.
+        ******************************************************************************
```

Comment:
...this will be indented by 8 spaces when output, different than the 4 previously used.

Probably best to use line concatenation or to join array elements with `\n`, ala:

```js
        [
          `******************************************************************************`,
          `Failed to deploy your app to version ${newVersionNum}. No changes were made.`,
          `Hostname: ${hostname}, Region: ${region}, Username: ${username}`,
          `To see the logs for your app, visit ${galaxyUrl}/logs.`,
          `******************************************************************************`,
        ].join("\n");
```

Reply:
But to try to be a bit more concise, and avoid the banner `****`, could we just count on the fact that `hostname`, `username`, etc, have already been output above and just say:

    Error: Failed to deploy new app version ${newVersionNum}. See ${galaxyUrl}/logs for details.

Reply:
Not sure if they already have been output. But I agree that concise is nice.

Reply:
I figured it would be easier to modify in the future if left as one big string, but I see the motivation behind .join("\n"). I also found the repetition of hostname, region, etc. useful for myself just as a sanity check to make sure you're deploying in the right region with the right hostname, etc.
The banner was just for a clear indication that the deployment succeeded. After waiting up to 10 minutes, I'd be happy to have a big banner dumped onto the screen that makes it very clear deploy has succeeded. It would also be much _much_ easier to find inside of CI logs, which is where we expect people to use this synchronous deployment feature. 

For now I'm gonna make the Failure message more concise (and add "ERROR" in front of it), but leave the success message as is. If you feel strongly about the banner, then lemme know, but I don't imagine it'll bother anyone, and it might help a few people save a minute or two when combing CI logs.

Reply:
Yeah, I think it's a matter of style. I tend to be inclined to more of an old-school Unix aesthetic, where programs are moderately concise on success and try to get your attention more on errors. I recognize that folks are more into colors and emoji and such in the terminal these days.

I'm also honestly a bit skeptical of being sure that the ***s will properly get through the wrapping that Meteor does and end up consistently on their own line...

Perhaps @abernix or @benjamn are a better arbiter of meteor CLI style than I am.